certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -133,6 +167,71 @@ def unsaved_files(self):  # pragma: no cover
         """Returns a list of unsaved filepaths"""
         return [assertions.PASS]
 
+    def _create_directivenode(self, path):
+        """Helper function to create a DirectiveNode from Augeas path"""
+
+        name = self.parser.get_arg(path)
+        params = tuple(self._aug_get_params(path))
+        metadata = {"augeasparser": self.parser, "augeaspath": path}
+
+        # Because of the dynamic nature, and the fact that we're not populating
+        # the complete ParserNode tree, we use the search parent as ancestor
+        return AugeasDirectiveNode(name=name,
+                                   parameters=params,
+                                   ancestor=self,
```

Comment:
Creating a fake tree with a different internal structure completely different from the actual AST seems like it might bite us in the future. Is the reason you don't set it to `None` because `None` would imply it's a root node?